### ISSUE TITLE
sshd-filter: optional port part

### ISSUE THREAD

**Issue by sebres** [fail2ban/fail2ban#1533]
see comments below

``` diff
- ^%(__prefix_line)s[iI](?:llegal|nvalid) user .* from <HOST>\s*$
+ ^%(__prefix_line)s[iI](?:llegal|nvalid) user .*? from <HOST>(?:\s+port\s+\d+)?\s*$
```

Note the extensions from #1209 (common sshd, sshd-ddos, sshd-aggressive)
It may be possible to unite it all, using options for filter...

**Comment by sebres** [comment]
<sub>moved from another not-related issue)</sub>
> @NoTeefy 

Fail2Ban stopped recognizing login attempts when I updated it to the 9.5.0 version.
My jails look like this:
```
[sshd]

enabled  = true
port    = ssh
filter   = sshd
action   = iptables[name=SSH, port=ssh, protocol=tcp]
           blocklist_de[email="xxx@xxx.xx", apikey="xxxx", service=%(filter)s]
logpath = %(sshd_log)s
backend = systemd
maxretry = 3
findtime = 86400

[sshd-ddos]
# This jail corresponds to the standard configuration in Fail2ban.
# The mail-whois action send a notification e-mail with a whois request
# in the body.
enabled  = true
port    = ssh
filter   = sshd-ddos
action   = iptables[name=SSH, port=ssh, protocol=tcp]
           blocklist_de[email="xxx@xxx.xx", apikey="xxxx", service=%(filter)s]
logpath = %(sshd_log)s
backend = systemd
maxretry = 3
findtime = 86400
```

And from the logging side I get logs like this:
```
2016-08-30 20:40:07,715 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:21.063236 hostname sshd[6467]: Disconnected from xx.xxx.xxx.xxx port 46349 [preauth]'
2016-08-30 20:40:07,715 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:21.085503 hostname sshd[6469]: Connection from xx.xxx.xxx.xxx port 53353 on xxx.xxx.xxx.xxx port 22'
2016-08-30 20:40:07,715 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:23.120815 hostname sshd[6469]: Invalid user operator from xx.xxx.xxx.xxx port 53353'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:23.120909 hostname sshd[6469]: input_userauth_request: invalid user operator [preauth]'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:25.381255 hostname sshd[6469]: Received disconnect from xx.xxx.xxx.xxx port 53353:11:  [preauth]'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:25.381268 hostname sshd[6469]: Disconnected from xx.xxx.xxx.xxx port 53353 [preauth]'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:25.404018 hostname sshd[6474]: Connection from xx.xxx.xxx.xxx port 53734 on xxx.xxx.xxx.xxx port 22'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:26.246909 hostname sshd[6474]: Invalid user webadmin from xx.xxx.xxx.xxx port 53734'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:26.247016 hostname sshd[6474]: input_userauth_request: invalid user webadmin [preauth]'
2016-08-30 20:40:07,716 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:34:28.423240 hostname sshd[6474]: Received disconnect from xx.xxx.xxx.xxx port 53734:11:  [preauth]'
2016-08-30 20:46:54,519 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.097943 hostname sshd[7634]: Connection from xx.xxx.xxx.xxx port 1112 on xxx.xxx.xxx.xxx port 22'
2016-08-30 20:46:54,519 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.097943 hostname sshd[7634]: Connection from xx.xxx.xxx.xxx port 1112 on xxx.xxx.xxx.xxx port 22'
2016-08-30 20:46:55,019 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.582124 hostname sshd[7634]: Received disconnect from xx.xxx.xxx.xxx port 1112:11: Bye Bye [preauth]'
2016-08-30 20:46:55,019 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.582124 hostname sshd[7634]: Received disconnect from xx.xxx.xxx.xxx port 1112:11: Bye Bye [preauth]'
2016-08-30 20:46:55,019 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.582136 hostname sshd[7634]: Disconnected from xx.xxx.xxx.xxx port 1112 [preauth]'
2016-08-30 20:46:55,020 fail2ban.filtersystemd  [4203]: DEBUG   Read systemd journal entry: '2016-08-30T20:46:54.582136 hostname sshd[7634]: Disconnected from xx.xxx.xxx.xxx port 1112 [preauth]'
```

If I run the **fail2ban-regex "systemd-journal" /etc/fail2ban/filter.d/sshd.conf** command it gives me back: 
```
Running tests
=============

Use   failregex filter file : sshd, basedir: /etc/fail2ban
Use         maxlines : 10
Use    journal match : _SYSTEMD_UNIT=sshd.service + _COMM=sshd


Results
=======

Failregex: 0 total

Ignoreregex: 0 total

Lines: 613 lines, 0 ignored, 0 matched, 613 missed
[processed in 0.34 sec]

Missed line(s): too many to print.  Use --print-all-missed to print all 613 lines
```
What is the problem right here? Can't find any errors even with fail2ban DEBUG logging on.

Best regards

**Comment by sebres** [comment]
<sub>moved from another not-related issue)</sub>
> @NoTeefy:

How can I fix those problems? I don't get any errors even with loglevel=debug . It just reads those failed attempts but doesn't do anything about it... Oh and by the way I just checked the log and when I'm starting Fail2Ban it shows me this output:
```
2016-08-30 22:21:17,552 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd
iptables -w -A f2b-sshd -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd
2016-08-30 22:21:17,553 fail2ban.jail           [21667]: INFO    Jail 'sshd' started
2016-08-30 22:21:17,553 fail2ban.transmitter    [21667]: DEBUG   Command: ['start', 'sshd-ddos']
2016-08-30 22:21:17,553 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd-ddos
iptables -w -A f2b-sshd-ddos -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd-ddos
2016-08-30 22:21:17,554 fail2ban.jail           [21667]: INFO    Jail 'sshd-ddos' started
2016-08-30 22:21:17,554 fail2ban.transmitter    [21667]: DEBUG   Command: ['start', 'named-refused']
2016-08-30 22:21:17,554 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-named-refused
iptables -w -A f2b-named-refused -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports domain,953 -j f2b-named-refused
2016-08-30 22:21:17,554 fail2ban.filterpyinotify[21667]: DEBUG   pyinotifier started for named-refused.
2016-08-30 22:21:17,554 fail2ban.jail           [21667]: INFO    Jail 'named-refused' started
2016-08-30 22:21:17,656 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd
iptables -w -A f2b-sshd -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd -- stdout: b''
2016-08-30 22:21:17,656 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd
iptables -w -A f2b-sshd -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd -- stderr: b''
2016-08-30 22:21:17,657 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd
iptables -w -A f2b-sshd -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd -- returned successfully
2016-08-30 22:21:17,658 fail2ban.action         [21667]: DEBUG   
2016-08-30 22:21:17,658 fail2ban.action         [21667]: DEBUG   Nothing to do
2016-08-30 22:21:17,758 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd-ddos
iptables -w -A f2b-sshd-ddos -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd-ddos -- stdout: b''
2016-08-30 22:21:17,758 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd-ddos
iptables -w -A f2b-sshd-ddos -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd-ddos -- stderr: b''
2016-08-30 22:21:17,759 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-sshd-ddos
iptables -w -A f2b-sshd-ddos -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd-ddos -- returned successfully
2016-08-30 22:21:17,759 fail2ban.action         [21667]: DEBUG   
2016-08-30 22:21:17,760 fail2ban.action         [21667]: DEBUG   Nothing to do
2016-08-30 22:21:17,860 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-named-refused
iptables -w -A f2b-named-refused -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports domain,953 -j f2b-named-refused -- stdout: b''
2016-08-30 22:21:17,860 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-named-refused
iptables -w -A f2b-named-refused -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports domain,953 -j f2b-named-refused -- stderr: b''
2016-08-30 22:21:17,860 fail2ban.action         [21667]: DEBUG   iptables -w -N f2b-named-refused
iptables -w -A f2b-named-refused -j RETURN
iptables -w -I INPUT -p tcp -m multiport --dports domain,953 -j f2b-named-refused -- returned successfully
```
I think this shouldn't be a problem because every jail gets **returned successfully** at the end. Those are the only strange loggings, I just get read systemd *logline* but not a single action is being triggered.
My guess would be the ports after the messages. Before I updated my system no ports were written in my logs and after it, it always prints out the port with it. If I pass the regex an auth-file from a week ago (without any ports) it passes them correctly. How can I disable the port logging?

Best regards

**Comment by sebres** [comment]
<sub>moved from another not-related issue)</sub>
> @NoTeefy:

Thanks! That fixed it. Where else do I need to add this part? Are you going to change it globally in a new version?

**EDIT:** 
Okay, so apparently it should be changed in sshd-ddos too. So from

`failregex = ^%(__prefix_line)sDid not receive identification string from <HOST>\s*$`

to
`
failregex = ^%(__prefix_line)sDid not receive identification string from <HOST>(?:\s+port\s+\d+)?\s*$`

Best regards

**Comment by sebres** [comment]
fixed with #1581
